editoast/src/views/v2/scenario.rs

```diff
@@ -0,0 +1,629 @@
+        let fixtures = scenario_v2_fixture_set.await;
+
+        let url = scenario_url(
+            fixtures.project.id(),
+            fixtures.study.id(),
+            Some(fixtures.scenario.id()),
+        );
+
+        let request = TestRequest::get().uri(&url).to_request();
+        let response: ScenarioResponseV2 = call_and_read_body_json(&service, request).await;
+
+        assert_eq!(response.id, fixtures.scenario.id());
+        assert_eq!(response.name, fixtures.scenario.model.name);
+    }
+
+    #[rstest]
+    async fn get_scenarios(#[future] scenario_v2_fixture_set: ScenarioV2FixtureSet) {
+        let service = create_test_service().await;
+        let fixtures = scenario_v2_fixture_set.await;
+
+        let url = scenario_url(fixtures.project.id(), fixtures.study.id(), None);
+
+        let request = TestRequest::get().uri(&url).to_request();
+        let response: PaginatedResponse<ScenarioWithDetailsV2> =
+            call_and_read_body_json(&service, request).await;
+
+        assert!(!response.results.is_empty());
+    }
+
+    #[rstest]
+    async fn get_scenarios_with_wrong_study(
+        #[future] scenario_v2_fixture_set: ScenarioV2FixtureSet,
+    ) {
+        let service = create_test_service().await;
+        let fixtures = scenario_v2_fixture_set.await;
+
+        let url = scenario_url(
+            fixtures.project.id(),
+            99999999,
+            Some(fixtures.scenario.id()),
+        );
+
+        let request = TestRequest::get().uri(&url).to_request();
+        let response = call_service(&service, request).await;
+
+        assert_eq!(response.status(), StatusCode::NOT_FOUND);
+    }
+
+    #[rstest]
+    async fn post_scenario(
+        #[future] scenario_v2_fixture_set: ScenarioV2FixtureSet,
+        db_pool: Data<DbPool>,
+    ) {
+        let service = create_test_service().await;
+        let fixtures = scenario_v2_fixture_set.await;
+
+        let timetable = Timetable::changeset()
+            .electrical_profile_set_id(None)
+            .create(&mut db_pool.get().await.unwrap())
+            .await
+            .unwrap();
+
+        let url = scenario_url(fixtures.project.id(), fixtures.study.id(), None);
+
+        let study_name = "new created scenario V2";
+        let study_description = "new created scenario description V2";
+        let study_timetable_id = timetable.id();
+        let study_infra_id = fixtures.infra.id();
+        let study_tags = Tags::new(vec!["tag1".to_string(), "tag2".to_string()]);
+
+        // Insert scenario
+        let request = TestRequest::post()
+            .uri(&url)
+            .set_json(json!({
+                "name": study_name,
+                "description": study_description,
+                "infra_id": study_infra_id,
+                "timetable_id": study_timetable_id,
+                "tags": study_tags
+            }))
+            .to_request();
+        let response: ScenarioResponseV2 = call_and_read_body_json(&service, request).await;
```

Comment:
Likewise, create a scenario fixture before asserting.

Reply:
I'm testing the creation of scenario, can't really create it before, but i can delete the scenario from database before asserting and it will be deleted even if asserts fails